### ISSUE TITLE
Add an expired property

### ISSUE THREAD

**Issue by asvetlov** [aio-libs/async-timeout#11]
1. Context manager should return itself from `__enter__`.
2. `expired` property should return `True` if timeout expired.

The change allows to figure out is `asyncio.TimeoutError` was raised by context manager itself or internal code, e.g.
```
try:
    async with timeout(1.0) as cm:
        await fetch()
except asyncio.TimeoutError:
    if cm.expired:
         # top level timeout
    else:
         # `fetch()` raised timeout itself
```

@hellysmile  we discussed the change offline.
Do you agree with proposed API?

**Comment by hellysmile** [comment]
Looks perfect. Another way is just `bool(cm)`, not sure is it clear or nor

**Comment by asvetlov** [comment]
Explicit property is better I believe.

**Comment by asvetlov** [comment]
Fixed by #16
